internal/datastore/mysql/migrations/driver.go

```diff
@@ -98,16 +100,52 @@ func (driver *MySQLDriver) Version(ctx context.Context) (string, error) {
 	return "", errors.New("no migration version detected")
 }
 
-// WriteVersion overwrites the _meta_version_ column name which encodes the version
+func (driver *MySQLDriver) Conn() Wrapper {
+	return Wrapper{DB: driver.db, Tables: driver.tables}
+}
+
+func (driver *MySQLDriver) RunTx(ctx context.Context, f migrate.TxMigrationFunc[TxWrapper]) error {
+	return BeginTxFunc(
+		ctx,
+		driver.db,
+		&sql.TxOptions{Isolation: sql.LevelSerializable},
+		func(tx *sql.Tx) error {
+			return f(ctx, TxWrapper{tx, driver.tables})
+		},
+	)
+}
+
+// BeginTxFunc is a polyfill for database/sql which does a closure style transaction lifecycle.
+// The underlying transaction is aborted if the supplied function returns an error.
+// The underlying transaction is committed if the supplied function returns nil.
+func BeginTxFunc(ctx context.Context, db *sql.DB, txOptions *sql.TxOptions, f func(*sql.Tx) error) error {
+	tx, err := db.BeginTx(ctx, txOptions)
+	if err != nil {
+		return err
+	}
+	defer LogOnError(ctx, tx.Rollback)
```

Comment:
Is it okay to call Rollback after Commit has completed?

Reply:
Yes.